src/render/render.c

```diff
@@ -36,52 +38,66 @@ void	render_scene(t_minirt *mrt)
 		pos.x = 0;
 		while (pos.x < WIN_WIDTH)
 		{
-			ray.direction = ray_tracer(camera, pos, ratio);
-			intercept(mrt, pos, ray);
+			ray.direction = primary_ray(camera, pos, ratio);
+			ray_tracer(mrt, &ray);
+			refresh_buffer(mrt, &ray, pos);
 			pos.x++;
 		}
 		pos.y++;
 	}
+	mrt->count++;
 	mlx_put_image_to_window(mlx->mlx_ptr, mlx->win_ptr, mlx->img_ptr, 0, 0);
 }
 
-static inline void	intercept(t_minirt *mrt, t_vector2 pos, t_ray ray)
+static inline t_fvector3	primary_ray(t_camera *cam,
+	t_vector2 pos, float ratio)
+{
+	t_fvector3	ndc_vec;
+
+	ndc_vec = (t_fvector3){
+		-(2.0f * (((float)pos.x + 0.5f) / WIN_WIDTH) - 1.0f) * ratio,
+		-(2.0f * (((float)pos.y + 0.5f) / WIN_HEIGHT) - 1.0f)
+		* cam->iplane_scale, 1.0f};
+	return (ft_fnormalize(
+			ft_fvector3_sum(
+				ft_fvector3_sum(
+					(t_fvector3){cam->right.x * ndc_vec.x,
+					cam->right.y * ndc_vec.x, cam->right.z * ndc_vec.x},
+				(t_fvector3){cam->up.x * ndc_vec.y,
+				cam->up.y * ndc_vec.y, cam->up.z * ndc_vec.y}),
+		cam->normal)));
+}
+
+static inline void	ray_tracer(t_minirt *mrt, t_ray *ray)
 {
 	t_object	*cur;
 	void		(*render)(t_minirt *, t_ray *, t_object *);
-	t_mlx		*mlx;
 
-	ray.dist = 3.4E+38;
-	ray.color = ft_rgb(0, 0, 0);
+	ray->dist = 3.4E+38;
 	cur = mrt->objects;
-	mlx = mrt->mlx;
 	while (cur)
 	{
 		render = cur->render;
 		if (render)
-			render(mrt, &ray, cur);
+			render(mrt, ray, cur);
 		cur = cur->next;
 	}
-	*((unsigned int *)(mlx->data + (pos.y * mlx->ll + pos.x * mlx->cl)))
-		= (ray.color.r << 16 | ray.color.g << 8 | ray.color.b);
+	if (ray->dist >= 3.4E+37)
+		ray->color = (t_rgb){0, 0, 0};
```

Comment:
```suggestion
		ray->color = (t_rgb){0, 255, 0};
```
miror return green color 🤡